webonary-cloud-api/lambda/browseEntries.ts

```diff
@@ -20,37 +34,84 @@ export async function handler(
+    const pageLimit = Math.min(
+      Math.max(Number(event.queryStringParameters?.pageLimit ?? DB_MAX_DOCUMENTS_PER_CALL), 1),
+      DB_MAX_DOCUMENTS_PER_CALL,
+    );
 
-    let errorMessage = '';
     if (!text) {
-      errorMessage = 'Browse letter head must be specified.';
+      return callback(null, Response.badRequest('Browse letter head must be specified.'));
+    }
+
+    // set up main search
+    const dbFind: DbFindParameters = { dictionaryId };
+    if (lang) {
+      dbFind.reversalLetterHeads = { lang, value: text };
+    } else {
+      dbFind.letterHead = text;
     }
 
-    if (errorMessage) {
-      return callback(null, Response.badRequest(errorMessage));
+    // return count only
+    if (countTotalOnly && countTotalOnly === '1') {
+      const count = await db.collection(DB_COLLECTION_ENTRIES).countDocuments(dbFind);
+      return callback(null, Response.success({ count }));
     }
 
-    const dbFind: DbFindParameters = {};
-    dbFind.dictionaryId = dictionaryId;
+    // set up to return entries
+    let entries: EntryData[];
+    let dbSortKey: string;
+    let dbLocale = DB_COLLATION_LOCALE_DEFAULT;
+    const dbSkip = getDbSkip(pageNumber, pageLimit);
 
-    if (text) {
-      if (lang) {
-        dbFind.reversalLetterHeads = { lang, value: text };
-      } else {
-        dbFind.letterHead = text;
+    if (lang) {
+      // TODO: Include reversal language in sorting?
+      /*
+      dbSortKey = PATH_TO_ENTRY_MAIN_HEADWORD_VALUE;
+      if (DB_COLLATION_LOCALES.includes(lang)) {
+        dbLocale = lang;
+      }
+      */
+      entries = await db
+        .collection(DB_COLLECTION_ENTRIES)
+        .aggregate(
+          [
+            { $match: dbFind },
+            { $unwind: `$${PATH_TO_ENTRY_DEFINITION}` },
+            { $match: { [PATH_TO_ENTRY_DEFINITION_LANG]: lang } },
+            { $sort: { [PATH_TO_ENTRY_DEFINITION_VALUE]: 1 } },
+          ],
+          { collation: { locale: dbLocale, strength: DB_COLLATION_LOCALE_STRENGTH } },
+        )
+        .skip(dbSkip)
+        .limit(pageLimit)
+        .toArray();
+    } else {
+      // TODO: Make sure to set default sort for entries to be on main headword browse letter and value
+      dbSortKey = PATH_TO_ENTRY_MAIN_HEADWORD_VALUE;
+      if (mainLang && mainLang !== '' && DB_COLLATION_LOCALES.includes(mainLang)) {
+        dbLocale = mainLang;
       }
+      entries = await db
+        .collection(DB_COLLECTION_ENTRIES)
+        .find(dbFind)
+        .collation({ locale: dbLocale, strength: DB_COLLATION_LOCALE_STRENGTH })
+        .sort({ [dbSortKey]: 1 })
+        .skip(dbSkip)
+        .limit(pageLimit)
+        .toArray();
     }
 
-    const entries: EntryData[] = await db
-      .collection(DB_COLLECTION_ENTRIES)
-      .find(dbFind)
-      .toArray();
-
     if (!entries.length) {
       return callback(null, Response.notFound([{}]));
     }
 
+    /* Manual sorting, if necessary
```

Comment:
I don't usually like to leave commented out code for very long. I'd at least put a TODO or some type of comment that would grab your attention in VSCode so that it would annoy someone enough to realize we can delete it if we don't need it.

Reply:
made it a function in common db.ts. I anticipate that we might need this for special sorting, when sorting via mongo is not good enough